bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/i18n/I18nUtil.java

```diff
@@ -12,21 +12,43 @@
  */
 package org.eclipse.smarthome.core.i18n;
 
+import java.util.function.Supplier;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
 /**
  *
  * @author Denis Nobel - Initial contribution
  */
+@NonNullByDefault
 public class I18nUtil {
 
     /** The 'text' pattern (prefix) which marks constants. */
     private static final String CONSTANT_PATTERN = "@text/";
 
-    public static boolean isConstant(String key) {
+    public static boolean isConstant(@Nullable String key) {
         return key != null && key.startsWith(CONSTANT_PATTERN);
     }
 
     public static String stripConstant(String key) {
         return key.replace(CONSTANT_PATTERN, "");
     }
 
+    /**
+     * If key is a constant strip the constant part, otherwise use the supplier provided string.
+     *
+     * @param key the key
+     * @param supplier the supplier that return value is used if key is identified as a constant
+     * @return the key with the stripped constant marker or the supplier provided key if it is not identified as a
+     *         constant
+     */
+    public static String stripConstantOr(final @Nullable String key, Supplier<String> supplier) {
+        if (key != null && key.startsWith(CONSTANT_PATTERN)) {
```

Comment:
why not use `isConstant` here?

Reply:
`stripConstant` needs to be called with a non-null argument.
The Eclipse Nullness support we are using cannot express "if isConstant returns true the given argument is known to be non-null" (the Checker Framework can handle such cases, e.g. https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/EnsuresNonNullIf.html).
The Eclipse Nullness support knows that if the check `key != null` in the same function has been passed, that key is non-null.